### PR TITLE
Adds proxy server support.

### PR DESCRIPTION
**Pull request by petelacey**

### CODE REVIEW

**Comment by petelacey** [comment]
Bump.  Any feedback on this PR?

File: lib/rollbar/notifier.rb [hunk 1]
```diff
@@ -496,6 +500,40 @@ def do_post(uri, body, access_token)
...
+      px.password = proxy[:password]
+      px
+    end
+
+    def proxy_from_env(uri)
+      uri.find_proxy
+    end
+
+    def null_proxy
+      OpenStruct.new(:host => nil, :port => nil, :user => nil, :password => nil)
```
Comment (rokob) [inline lib/rollbar/notifier.rb]: Why OpenStruct and not just a plain struct? i.e.

`Struct.new(:host, :port, :user, :password).new`

Comment (petelacey) [inline lib/rollbar/notifier.rb]: To be honest, this started life as a Struct, then an abandoned refactoring changed it to an OpenStruct and I neglected to change it back.  Happy to do that.  It makes this code more concise.

**Comment by rokob** [comment]
This seems like a reasonable change to me.

File: lib/rollbar/notifier.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@
 require 'rollbar/delay/thread'
 require 'rollbar/logger_proxy'
 require 'rollbar/item'
+require 'ostruct'
```
Comment (printercu) [inline lib/rollbar/notifier.rb]: This one is redundant.

**Pull request merged**